src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/PartiallyDeterminedHaplotypeComputationEngine.java

```diff
@@ -638,131 +622,138 @@ static PartiallyDeterminedHaplotype createNewPDHaplotypeFromEvents(final Haploty
-            //TODO using the bitmask values themselves for the loop
-            if (!forbiddenCombinations.isEmpty()) {
-                for (final SmallBitSet subset = new SmallBitSet().increment(); !subset.hasElementGreaterThan(eventsInOrder.size()); subset.increment()) {
-                    if (forbiddenCombinations.stream().anyMatch(subset::contains)) {
-                        allowedEvents.set(subset.index(), false);
-                    }
-                }
-            }
-        }
+             // make SmallBitSet of the event indices of each mutex
+             final List<SmallBitSet> mutexes = mutexPairsAndTrios.stream().map(mutex -> new SmallBitSet(mutex.stream().map(eventIndices::get).toList())).toList();
+
+             // Now forbid all subsets that contain forbidden combinations
+             //TODO This method is potentially very inefficient! We don't technically have to iterate over every i,
+             //TODO we know there is an optimization involving minimizing the number of checks necessary here by iterating
+             //TODO using the bitmask values themselves for the loop
+             if (!mutexes.isEmpty()) {
+                 for (final SmallBitSet subset = new SmallBitSet().increment(); !subset.hasElementGreaterThan(eventsInOrder.size()); subset.increment()) {
+                     if (mutexes.stream().anyMatch(subset::contains)) {
+                         allowedSubsets.set(subset.index(), false);
+                     }
+                 }
+             }
+         }
 
         /**
-         * This method handles the logic involved in getting all of the allowed subsets of alleles for this event group.
+         * For a given set of determined events, find all subsets of the event group such that each subset
+         *
+         * 1) contains all determined events that overlap this event group
+         * 2) contains no two or three mutually exclusive elements i.e. is an allowed subset as computed in this object's constructor
+         * 3) is not a subset of any larger subset that satisfies 1) and 2)
+         *
+         * Note that the subsets need not be disjoint and need not cover all events in this EventGroup.
+         *
+         * Because these subsets contain no mutexes, one can sensibly make partially-determined haplotypes out of them,
+         * with each subset entailing a unique set of undetermined alleles.  Property 3) above ensures that we make as
+         * few partially-determined haplotypes as possible.
          *
-         * @param disallowSubsets
          * @return
          */
-        public List<Set<Event>> setsForBranching(final List<Event> locusEvents, final Set<Event> determinedEvents, final boolean disallowSubsets) {
+        public List<Set<Event>> eventSetsForPDHaplotypes(final Set<Event> determinedEvents, final List<Event> locusEvents) {
             final SmallBitSet locusOverlapSet = overlapSet(locusEvents);
```

Comment:
consider overlapSet -> intersectVsEventGroup to make it a little more legible what is happening here.

Reply:
or perhaps just a comment